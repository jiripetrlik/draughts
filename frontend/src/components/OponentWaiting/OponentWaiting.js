import React from 'react';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

class OponentWaiting extends React.Component {

    render() {
        return (
            <div className="oponent-waiting">
                <Typography>Waiting for oponent</Typography>
                <Button>Cancel</Button>
            </div>
        )
    }
}

export default OponentWaiting
