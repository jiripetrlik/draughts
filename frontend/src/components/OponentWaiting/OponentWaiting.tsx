import React from 'react';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { sendMessage } from '../../api/index'

class OponentWaiting extends React.Component {

    stopWaiting() {
        let command = {
            Name: "stop-waiting",
            Parameters: ""
        }

        sendMessage(JSON.stringify(command))
    }

    render() {
        return (
            <div className="oponent-waiting">
                <Typography>Waiting for oponent</Typography>
                <Button onClick={this.stopWaiting}>Cancel</Button>
            </div>
        )
    }
}

export default OponentWaiting
