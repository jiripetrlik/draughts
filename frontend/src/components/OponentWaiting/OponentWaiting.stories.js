import React from 'react';
import OponentWaiting from './OponentWaiting'

export default {
  component: OponentWaiting,
  title: 'OponentWaiting',
};

const Template = args => <OponentWaiting {...args} />;

export const Default = Template.bind({});
