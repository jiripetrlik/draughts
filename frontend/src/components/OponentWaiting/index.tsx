import OponentWaiting from "./OponentWaiting"

export default OponentWaiting
