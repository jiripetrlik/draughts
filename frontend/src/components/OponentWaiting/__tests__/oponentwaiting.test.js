import React from 'react';
import renderer from 'react-test-renderer';
import OponentWaiting from '../OponentWaiting'

test('snapshot test', () => {
    const tree = renderer
        .create(<OponentWaiting />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
