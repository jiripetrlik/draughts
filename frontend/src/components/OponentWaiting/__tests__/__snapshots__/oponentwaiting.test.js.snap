// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="oponent-waiting"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    Waiting for oponent
  </p>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Cancel
    </span>
  </button>
</div>
`;
